core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -704,7 +705,27 @@ public int hashCode() {
         /**
          * Returns the next operation in the snapshot or <code>null</code> if we reached the end.
          */
-        Translog.Operation next() throws IOException;
+        Position next() throws IOException;
+
+    }
+
+    public static class Position {
```

Comment:
To feed the translog location down to the engine where it's used when modifying the version map.